src/test/java/io/odh/test/e2e/standard/PipelineServerST.java

```diff
@@ -0,0 +1,584 @@
+import io.fabric8.kubernetes.api.model.SecretBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.openshift.client.OpenShiftClient;
+import io.odh.test.OdhAnnotationsLabels;
+import io.odh.test.TestUtils;
+import io.odh.test.framework.listeners.ResourceManagerDeleteHandler;
+import io.odh.test.framework.manager.ResourceManager;
+import io.opendatahub.datasciencepipelinesapplications.v1alpha1.DataSciencePipelinesApplicationBuilder;
+import io.opendatahub.datasciencepipelinesapplications.v1alpha1.datasciencepipelinesapplicationspec.ApiServer;
+import lombok.SneakyThrows;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/// helpers for reading json responses
+/// there should be openapi spec, so this can be generated
+
+class PipelineResponse {
+    public List<Pipeline> pipelines;
+    public int total_size;
+}
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+class Pipeline {
+    public String id;
+    public String name;
+}
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+class ApiRunDetail {
+    public PipelineRun run;
+}
+
+class ApiListRunsResponse {
+    public List<PipelineRun> runs;
+    public int total_size;
+    public String next_page_token;
+}
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+class PipelineRun {
+    public String id;
+    public String name;
+    public PipelineSpec pipeline_spec;
+
+    public String created_at;
+    public String scheduled_at;
+    public String finished_at;
+    public String status;
+    public String error;
+}
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+class PipelineSpec {
+    public String pipeline_id;
+    public String pipeline_name;
+}
+
+/**
+ * Q1: which http library?
+ * fabric8 uses okhttp3
+ * there is HttpClient in java 17; does not have multipart form post request helper class?
+ * found https://github.com/rest-assured/rest-assured
+ * <p>
+ * Q2: how to deal with jsons from kf pipeline?
+ * fabric8 uses jackson; either use ad hoc, or generate client from openapi/swagger
+ * rest-assured can read json fields using string paths (example in its readme)
+ *
+ * try graalpython and simply use python client from java :P
+ * https://kf-pipelines.readthedocs.io/en/latest/source/kfp.client.html
+ */
+
+@ExtendWith(ResourceManagerDeleteHandler.class)
+public class PipelineServerST {
```

Comment:
can it extend standard abstract?

Reply:
after I do that, running the test will first try to install rhoai/rhods. can I skip that on my machine and use whatever rhoai I have already in the cluster?

Reply:
well it is easy to add env var switch to allow do not install operator.... instead of addidn continuous profile etc.